lib/node_types/esp/src/device-manager.h

```diff
@@ -1,7 +1,83 @@
-// device-manager.h
-// author: ulno
-// created: 2018-07-16
-// Manage a list of devices
+/**
+ * @file device-manager.h
+ * @brief Central manager for all devices and the do_later scheduler
+ * @author ulno
+ * @created 2018-07-16
+ * 
+ * DEVICEMANAGER - THE HEART OF IOTEMPOWER
+ * ========================================
+ * The DeviceManager is a singleton that manages all devices on a node.
+ * It's responsible for:
+ * - Device registration and tracking
+ * - Device lifecycle management (start, update, publish)
+ * - Coordinating MQTT operations across all devices
+ * - Managing the do_later scheduler
+ * 
+ * WHY DEVICEMANAGER IS ESSENTIAL
+ * ==============================
+ * In a standard Arduino project, you would need:
+ * 
+ *   Device* devices[10];
+ *   int device_count = 0;
+ *   
+ *   void setup() {
+ *     devices[device_count++] = &temp_sensor;
+ *     devices[device_count++] = &humidity_sensor;
+ *     devices[device_count++] = &led;
+ *     // Initialize each device
+ *     for(int i=0; i<device_count; i++) {
+ *       devices[i]->start();
+ *     }
+ *   }
+ *   
+ *   void loop() {
+ *     // Poll each device
+ *     for(int i=0; i<device_count; i++) {
+ *       devices[i]->poll();
+ *     }
+ *     // Publish each device
+ *     for(int i=0; i<device_count; i++) {
+ *       devices[i]->publish();
+ *     }
+ *   }
+ * 
+ * With IoTempower's DeviceManager:
+ *   // Just create devices - that's it!
+ *   Temperature temp_sensor("temperature");
```

Comment:
The device creation should not be done as described (compare docs/node_help or architecture or examples).
It would be something like:
dht11(temp_hum, D4); // creates device for topic "<node_folder_name>/temp_hum" and publishing on subtopics temperature and humidity
output(light, D1); // create device listening on <node_folder_name>/light/set and reporting on <node_folder_name>/light

helper macros are setting the quotation marks themselves and using the respective classes.